src/main/scala/viper/gobra/translator/encodings/FloatEncoding.scala

```diff
@@ -0,0 +1,270 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2021 ETH Zurich.
+
+package viper.gobra.translator.encodings
+
+import org.bitbucket.inkytonik.kiama.==>
+import viper.gobra.ast.{internal => in}
+import viper.gobra.theory.Addressability
+import viper.gobra.theory.Addressability.{Exclusive, Shared}
+import viper.gobra.translator.interfaces.{Collector, Context}
+import viper.gobra.translator.util.ViperWriter.CodeLevel.unit
+import viper.gobra.translator.util.ViperWriter.CodeWriter
+import viper.silver.{ast => vpr}
+
+class FloatEncoding extends LeafTypeEncoding {
+
+  import viper.gobra.translator.util.TypePatterns._
+
+  /**
+    * Translates a type into a Viper type.
+    */
+  override def typ(ctx: Context): in.Type ==> vpr.Type = {
+    case ctx.Float32() / Exclusive => floatType32
+    case ctx.Float64() / Exclusive => floatType64
+    case ctx.Float32() / Shared => vpr.Ref
+    case ctx.Float64() / Shared => vpr.Ref
+  }
+
+  private lazy val floatType32: vpr.Type = {
+    isUsed32 = true
+    vpr.Int
+  }
+
+  private lazy val floatType64: vpr.Type = {
+    isUsed64 = true
+    vpr.Int
+  }
+
+  /**
+    * Encodes expressions as values that do not occupy some identifiable location in memory.
+    * X stands for either 32 or 64 below:
+    * [ dflt(x: floatX) ] -> [ floatX(0) ]
+    * [ (x: floatX) + (y: floatX) ] -> addFloatX([ x ], [ y ])
+    * [ (x: floatX) - (y: floatX) ] -> subFloatX([ x ], [ y ])
+    * [ (x: floatX) * (y: floatX) ] -> mulFloatX([ x ], [ y ])
+    * [ (x: floatX) / (y: floatX) ] -> divFloatX([ x ], [ y ])
+    * [ floatX(x: int) ] -> fromIntToX([ x ])
+    */
+  override def expr(ctx: Context): in.Expr ==> CodeWriter[vpr.Exp] = {
```

Comment:
contains quite a bit of repetition. One could parameterize the `addFloat`, `subFloat`, etc. functions by the float size but I'm not sure whether that worth the code complexity